Check KV4 compatibility with attention backends and add KV4 support to the attention_backend doc

JackChuang opened the pull request:
## Motivation

Prevent users from using KV4 with incompatible attention backends by clearly documenting supported backends and enforcing runtime checks.

Improves reliability and reduces runtime errors by ensuring users cannot accidentally run KV4 with unsupported attention backends.

## Modifications

Ensure code executes after default settings are set by placing it in server_args.py instead of server_args.py.

Description / Changes:
	1.	Backend documentation updates
	•	Added FP4 KV cache column to the MLA/MHA backend table.
	•	Clarifies which backend combinations are supported with FP4 KV caches.
	2.	ServerArgs updates
	•	Added _handle_kv4_compatibility() to check KV4 compatibility with attention backends at runtime.
	•	Logs warnings for potential edge-case incompatibilities.
	•	Adds assertions to enforce correct decode_attention_backend for FA4 + MLA/MHA and non-FA4 + MLA/MHA setups.
	•	Raises an error if KV4 is used on non-CUDA platforms.

## Testing

The compatibility results are tested on B200 (sm100), using Qwen3-235B-A22B as MHA and DeepSeek-R1-0528-FP4 as MLA to test.

## Next (WIP)

Test kv4 with fa3 and flashmla backend on sm90 to complete the table. I will send another PR for this.


## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [x] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

JackChuang committed:
Add FP4 KV cache column to MHA and MLA backend table and update FA4 + FP4 note

- Introduce FP4 KV cache support in the backend matrix.
- Add note on FA4 + KV4 scenario.

Signed-off-by: Ho-Ren (Jack) Chuang <horenchuang@bytedance.com>

JackChuang committed:
Check KV4 compatibility with attention backends

- Add _handle_kv4_compatibility() to validate backend choices for KV4 scenarios.
- Warns on potential edge-case incompatibilities.
- Asserts correct decode_attention_backend for FA4 + MLA/MHA and non-FA4 + MLA/MHA setups.
- Raises error if KV4 is used on non-CUDA platforms.

Signed-off-by: Ho-Ren (Jack) Chuang <horenchuang@bytedance.com>

Fridge003 commented on python/sglang/srt/model_executor/model_runner.py:
```diff
@@ -384,6 +387,80 @@ def init_mindspore_runner(self):
                 port=self.dist_port,
             )
 
+    def _handle_kv4_compatibility(self):
```
Can we move this function to `server_args.py`, somewhere after `_handle_attention_backend_compatibility`

JackChuang replied:
No problem. I will fix this as soon as I can. Thank you for your feedback!

JackChuang commented:
@Fridge003 I've fixed the code according to your comments and I've also rebased to origin/main and fixed conflicts as well. Please check again. Thank you~

Fridge003 commented:
@JackChuang Please fix this https://github.com/sgl-project/sglang/actions/runs/20055608966/job/57520174252?pr=14467

JackChuang commented:
> @JackChuang Please fix this https://github.com/sgl-project/sglang/actions/runs/20055608966/job/57520174252?pr=14467

I've fixed the failing stage-a-test-1 test. Please check again. Thanks.
